h2o-algos/src/main/java/hex/rulefit/Rule.java

```diff
@@ -54,19 +52,18 @@ public void map(Chunk[] cs, byte[] out) {
             c.map(cs, out);
         }
     }
-    
-    @Override
-    public int hashCode() {
-        int hashCode = 0;
-        for (int i = 0; i < conditions.length; i++) {
-            hashCode += conditions[i].hashCode();
-        }
-        return hashCode;
-    }
 
     @Override
     public boolean equals(Object obj) {
-        return this.hashCode() == obj.hashCode();
+        if (!(obj instanceof Rule))
+            return false;
+
+        Rule rule = (Rule) obj;
+        for (int i = 0; i < conditions.length; i++) {
+            if (!(Arrays.stream(rule.conditions).anyMatch(conditions[i]::equals)))
```

Comment:
Can we create stream just once and use it multiple time in loop?

Reply:
No, stream can be consumed only once in Java. But I did some adjustments to make it nicer.